controller.py

```diff
@@ -215,6 +256,55 @@ def search(self, location, live=True):
             )
             return Response(body, 200, headers)
 
+    def libraries(self):
+        # Return a specific set of information about all libraries; this generates the library list in the admin interface,
+        libraries = []
+        all = self._db.query(Library).order_by(Library.name)
+        for library in all:
+            uuid = library.internal_urn.split("uuid:")[1]
+            libraries += [dict(
+                    id=library.id,
+                    name=library.name,
+                    short_name=library.short_name,
+                    uuid=uuid,
+                    registry_stage=library.registry_stage,
+                    library_stage=library.library_stage,
+                )]
+        return dict(libraries=libraries)
+
+    def library_details(self, uuid):
+        # Return complete information about one specific library; this is triggered when an admin opens the detail page for a library.
+        library = self.library_for_request(uuid)
+        if isinstance(library, ProblemDetail):
+            return library
+        library_info = dict(
+            name=library.name,
+            short_name=library.short_name,
+            authentication_url=library.authentication_url,
+            online_registration=library.online_registration,
+            description=library.description,
+            timestamp=library.timestamp,
+            internal_urn=library.internal_urn,
+            library_stage=library._library_stage,
+            opds_url=library.opds_url,
+            registry_stage=library.registry_stage,
+            web_url=library.web_url,
+        )
```

Comment:
It would be good to have contact emails in this info as well.

Reply:
Where would we be getting that information?  I'm not seeing anything in the Library model that looks like a contact email...

Reply:
I think they're Hyperlinks that are associated with the library.

Reply:
Yes, that seems right--thanks!